src/content/blog/biome-v1-9.md

```diff
@@ -0,0 +1,245 @@
+
+```json
+{
+  "css": {
+    "formatter": {
+      "enabled": false
+    },
+    "linter": {
+      "enabled": false
+    }
+  }
+}
+```
+
+or on the command line:
+
+```shell
+biome format --css-formatter-enabled=false
+biome lint --css-linter-enabled=false
+biome check --css-formatter-enabled=false --css-linter-enabled=false
+```
+
+Special thanks to [Denis Bezrukov @denbezrukov](https://github.com/denbezrukov) and [Jon Egeland @faultyserver](https://github.com/faultyserver) for implementing most of the CSS parser and formatter!
+Special thanks to [Yoshiaki Togami @togami2864](https://github.com/togami2864) for coordinating the implementation and implementing most of the CSS lint rules!
+
+### Stable GraphQL formatter and linter
+
+Biome now formats and lints GraphQL files by default.
+
+For now, Biome provides two nursery rules:
+
+- [nursery/noDuplicateFields](https://biomejs.dev/linter/rules/no-duplicate-fields/)
+- [nursery/useDeprecatedReason](https://biomejs.dev/linter/rules/use-deprecated-reason/)
+
+If you don't want Biome to format and lint your GraphQL files,
+you can disable the GraphQL formatter and linter in the Biome configuration file:
+
+```json
+{
+  "graphql": {
+    "formatter": {
+      "enabled": false
+    },
+    "linter": {
+      "enabled": false
+    }
+  }
+}
+```
+
+or on the command line:
+
+```shell
+biome format --graphql-formatter-enabled=false
+biome lint --graphql-linter-enabled=false
+biome check --graphql-formatter-enabled=false --css-linter-enabled=false
+```
+
+Special thanks to [Võ Hoàng Long @vohoanglong0107](https://github.com/vohoanglong0107) for implementing most of the features related to GraphQL.
+
+### Search command
+
+The `biome search` command allows you to search in your project using GritQL syntax.
+
+[GritQL](https://docs.grit.io/language/overview) is a powerful query language that lets you do structural searches on your codebase.
+This means that trivia such as whitespace or even the type of string quotes used will be ignored in your search query.
+It also has many features for querying the structure of your code, making it much more elegant for searching code than regular expressions.
+
+While we believe this command may already be useful to users in some situations (especially when integrated in the IDE extensions!), we also had an ulterior motive for adding this command:
+We intend to utilize GritQL for our plugin efforts, and by allowing our users to try it out in a first iteration, we hope to gain insight in the type of queries you want to do, as well as the bugs we need to focus on.
+
+For now, the search command is explicitly marked as **EXPERIMENTAL**, since many limitations are yet to be fixed or explored.
+Keep this in mind when you try it out, and please let us know what you think!
+
+Note: GritQL escapes code snippets using backticks, but most shells interpret backticks as command invocations. To avoid this, it's best to put single quotes around your Grit queries.
```

Comment:
```suggestion
Back in February, one of our Core Contributors published [a proposal for plugin support](https://github.com/biomejs/biome/discussions/1762). One of the highlights was the use of GritQL as a foundation for our plugin system.

[GritQL](https://docs.grit.io/language/overview) is a powerful query language that lets you do structural searches on your codebase. This means that trivia such as whitespace or even the type of string quotes used will be ignored in your search query. It also has many features for querying the structure of your code, making it much more elegant for searching code than regular expressions.

Integrating a query language such as GritQL is no easy feat, and throughout the year we published [multiple](https://github.com/biomejs/biome/discussions/2286) [status](https://github.com/biomejs/biome/discussions/2585) [updates](https://github.com/biomejs/biome/discussions/3392). Today, we release the first product of this effort: A new `biome search` command.

While we believe this command may already be useful to users in some situations (especially when it gets integrated in our IDE extensions!), this command is really a stepping stone towards our plugin efforts. By allowing our users to try it out in a first iteration, we hope to gain insight into the type of queries you want to do, as well as the bugs we need to focus on.

For now, the search command is explicitly marked as **EXPERIMENTAL**, since many limitations are yet to be fixed or explored. Keep this in mind when you try it out, and please let us know what you think!

Even though there are still plenty of limitations, we do believe the integration has progressed far enough that we can shift our focus towards the integration of actual plugins. We cannot yet promise a timeline, but we'll keep you posted!

PS.: GritQL escapes code snippets using backticks, but most shells interpret backticks as command invocations. To avoid this, it's best to put single quotes around your Grit queries.
```

Reply:
I think this screenshot might be nice to insert:

![biome_search_new](https://github.com/user-attachments/assets/30017a80-8b22-481c-bcb4-33b362b33f33)

Might be a nice fit right under the paragraph that ends with "Today, we release the first product of this effort: A new `biome search` command."

Reply:
@arendjr dummy question: why the middle one is a match? It doesn't contain `manifest`?

Reply:
This one might be a good alternative (I trimmed the results a bit for brevity, if you run this on our repo it will actually give some more results, but at least they're all correct):

```shell
$ biome search '`console.$method($args)` where { $method <: or { `info`, `warn` } }'
/home/arendjr/Projects/biome/packages/@biomejs/biome/scripts/postinstall.js:49:3 search ━━━━━━━━━━━━

  49 │          console.warn(
  50 │                  `The Biome CLI postinstall script failed to resolve the binary file "${binName}". Running Biome from the npm package will probably not work correctly.`,
  51 │          );

/home/arendjr/Projects/biome/benchmark/bench.js:183:3 search ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  183 │                 console.info(
  184 │                         `Adding directories ${dirs.join()} to sparse checkout in ${projectDirectory}`,
  185 │                 );

/home/arendjr/Projects/biome/benchmark/bench.js:193:2 search ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  193 │         console.info("Building Biome...");

Searched 24 files in 131ms. Found 3 matches.
```